http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -246,10 +250,20 @@ public void onNext(HttpContent httpContent) {
                     if (isDone.get()) {
                         return;
                     }
-                    // Needed to prevent use-after-free bug if the subscriber's onNext is asynchronous
-                    ByteBuffer b = copyToByteBuffer(httpContent.content());
-                    httpContent.release();
-                    subscriber.onNext(b);
+
+                    ByteBuffer byteBuffer = null;
+                    try {
+                        // Needed to prevent use-after-free bug if the subscriber's onNext is asynchronous
+                        byteBuffer = copyToByteBuffer(httpContent.content());
+                    } catch (Throwable throwable) {
+                        onError(throwable);
+                    } finally {
+                        httpContent.release();
+                    }
+
+                    if (byteBuffer != null) {
+                        subscriber.onNext(byteBuffer);
```

Comment:
should we still do a try catch here and complete the future in case we have a bad subscriber?

Reply:
+1 sounds good